### PR TITLE
Gallery Viewer Function

### PR DESCRIPTION
**Pull request by thaitwo**
@alexwolfe Can you please look over my gallery function when you have some time? 🍨 🍔  Thanks!

### CODE REVIEW

**Comment by alexwolfe** [comment]
Looks like you have some changes locally that you've haven't committed yet. Push those up and I'll review this today. Thanks.

File: public/stylesheets/viewer.css [hunk 1]
```diff
@@ -0,0 +1,77 @@
+.viewer {
+  position: fixed;
+  width: 100%;
+  height: 100%;
```
Comment (alexwolfe) [inline public/stylesheets/viewer.css]: You actually shouldn't use width or height in this class. the correct thing to do is: 
```
.viewer {
  position: fixed;
  top: 0; 
  left: 0; 
  right: 0; 
  bottom: 0; 
  z-index: 100;
  display: none;  
}
```
This pins all four corners of this div to the top, bottom, left, and right of the screen. When you position something absolutely or fixed that you want to cover the screen, this is the proper way to do it.

File: public/stylesheets/viewer.css [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  background: rgba(0, 0, 0, 1);
+  display: none;
+  z-index: 100;
+}
+
+.viewer.is-visible {
+  display: block;
+}
+
+.closeviewer {
```
Comment (alexwolfe) [inline public/stylesheets/viewer.css]: technically this should be
```
.viewer-close-button { 
```
This basically prefixes the items related (or inside the viewer) with the .viewer class. This reduces the chances of overwriting a style with a similar name somewhere else in your css.

File: public/stylesheets/viewer.css [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  color: #fff;
+}
+
+.closeviewer:hover {
+  color: #CFD8DC;
+  transition: .3s;
+  cursor: pointer;
+}
+
+.imageviewer {
```
Comment (alexwolfe) [inline public/stylesheets/viewer.css]: I'd recommend: 
```
.viewer-image-container { 
```

File: public/stylesheets/viewer.css [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+.imageviewer img {
+  max-width: 100%;
+  max-height: 100%;
+  display: block;
+  margin: auto;
+}
+
+.caption {
```
Comment (alexwolfe) [inline public/stylesheets/viewer.css]:
```
.viewer-caption { 
```

File: public/stylesheets/viewer.css [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  font: 16px "Raleway", sans-serif;
+  font-weight: 300;
+  letter-spacing: 0.05em;
+  display: block;
+  color: #fff;
+  text-align: center;
+  margin-top: 20px;
+}
+
+#prev, #next {
```
Comment (alexwolfe) [inline public/stylesheets/viewer.css]: These should be class not ids. you can have the ids on the element so javascript can find it, but I recommend creating a class for this. Example: 
```
.back-arrow, forward-arrow { 
```

File: routes/index.js [hunk 6]
```diff
@@ -36,4 +36,12 @@ router.get('/gallery/food', function(req, res, next) {
   res.render('gallery-food', { title: 'Food'});
 });
 
+/* GET freestyle page */
+router.get('/freestyle', function(req, res, next) {
```
Comment (alexwolfe) [inline routes/index.js]: If you're not longer using this, get rid of this route.

**Comment by alexwolfe** [comment]
@thaitwo nice work! Just had a few comments/suggestions for you. 👍

**Comment by thaitwo** [comment]
@alexwolfe Alright, I think everything is set. What do you think?

**Pull request merged**